lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/TaskStartExceptionTest.cs

```diff
@@ -0,0 +1,207 @@
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<TaskStartExceptionTestDriver>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<TaskStartExceptionTestDriver>.Class)
+                .Set(DriverConfiguration.OnTaskFailed, GenericType<TaskStartExceptionTestDriver>.Class)
+                .Build(), typeof(TaskStartExceptionTestDriver), 1, "testStartTaskWithExceptionOnLocalRuntime", "local", testFolder);
+
+            var driverMessages = new List<string>
+            {
+                FailedTaskReceived,
+                CompletedTaskReceived
+            };
+
+            ValidateMessagesSuccessfullyLoggedForDriver(driverMessages, testFolder, 1);
+
+            // Validate that the first Task never starts.
+            ValidateMessageSuccessfullyLogged(
+                new List<string> { InitialTaskMessage }, "Node-*", EvaluatorStdout, testFolder, 0);
+
+            ValidateMessageSuccessfullyLogged(
+                new List<string> { ResubmitTaskMessage }, "Node-*", EvaluatorStdout, testFolder, 1);
+
+            CleanUp(testFolder);
+        }
+
+        private sealed class TaskStartExceptionTestDriver :
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<ICompletedTask>,
+            IObserver<IFailedTask>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+
+            [Inject]
+            private TaskStartExceptionTestDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                // submit the first Task.
+                value.SubmitTask(TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, "TaskID")
+                        .Set(TaskConfiguration.Task, GenericType<TaskStartExceptionTask>.Class)
+                        .Set(TaskConfiguration.OnTaskStart, GenericType<TaskStartHandlerWithException>.Class)
+                        .Build());
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                // Should only receive one CompletedTask, as validated.
+                Logger.Log(Level.Info, CompletedTaskReceived);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                // Check that Exceptions are deserialized correctly.
+                var ex = value.AsError();
+                if (ex == null)
+                {
+                    throw new Exception("Exception was not expected to be null.");
+                }
+
+                var taskStartEx = ex as TaskStartExceptionTestException;
+
+                if (taskStartEx == null)
```

Comment:
It is Ok to throw exception here. You can also use Assert instead of exception as it is for test. The error message can be put inside Assert.